### PR TITLE
Add custom saved queries and fix duplicate requests

### PR DESCRIPTION
**Pull request by asteriscos**
### Description
This PR fixes automatic dashboard refresh and duplicate queries on the dashboard's initial load.
image
 
### Issues Resolved
[#7086](https://github.com/wazuh/wazuh/issues/7086)

### Evidence

https://github.com/user-attachments/assets/9246fc29-d12c-421c-8b20-b210f6b08d96

### Test
1. Remove the sample data, if present.
2. Add the sample data to obtain the latest timestamps.
3. Navigate to any application.
4. Set a refresh interval of 5 seconds within the time filter.
5. Observe whether the histogram updates over time.

> [!IMPORTANT]
> 1. Check if the URL updates with the savedQuery when you select an option or clear all (savedQueries)
> 2. Additionally, verify that the AppFilters are being saved correctly in the URL

https://github.com/user-attachments/assets/768486d4-2e5e-42e5-8436-d278b1fcdeb8

### Check List
- [x] All tests pass
  - [x] `yarn test:jest`
- [x] New functionality includes testing.
- [ ] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -25,6 +25,7 @@ All notable changes to the Wazuh app project will be documented in this file.
 - Fixed style when unnpinned an agent in endpoint summary section [#7015](https://github.com/wazuh/wazuh-dashboard-plugins/pull/7015)
 - Fixed overflow style on a long value filter [#7021](https://github.com/wazuh/wazuh-dashboard-plugins/pull/7021)
 - Fixed buttons enabled for a readonly user in `Endpoint groups` section [#7056](https://github.com/wazuh/wazuh-dashboard-plugins/pull/7056)
+- Fixed the automatic page refresh in dashboards and prevent duplicate requests [#7090](https://github.com/wazuh/wazuh-dashboard-plugins/pull/7090)
```
Comment (Desvelao) [inline CHANGELOG.md:29]: **thought**: we should add an entry related to the enhancement of performance or remove the duplicated requests in the dashboards/events.

File: plugins/main/public/components/common/search-bar/use-search-bar.ts [hunk 2]
```diff
@@ -126,6 +132,16 @@ const useSearchBarConfiguration = (
         : console.warn('setFilters function is not defined');
       props?.onFiltersUpdated && props?.onFiltersUpdated(userFilters);
     },
+    refreshInterval:
+      queryService.timefilter.timefilter.getRefreshInterval().value,
+    isRefreshPaused:
+      queryService.timefilter.timefilter.getRefreshInterval().pause,
+    onRefreshChange: opts => {
+      queryService.timefilter.timefilter.setRefreshInterval(opts);
```
Comment (Desvelao) [inline plugins/main/public/components/common/search-bar/use-search-bar.ts:140]: **nitpick**: the interface of `opts` has to be transformed. See the default method to manage this when the `useDefaultBehabiour` is enabled: https://github.com/opensearch-project/OpenSearch-Dashboards/blob/2.13.0/src/plugins/data/public/ui/search_bar/create_search_bar.tsx#L69-L72

But, I guess this does not update the absolute date range or should it update the date range if the date is relative?

File: CHANGELOG.md (hunk 1 above)
Comment (asteriscos) [reply CHANGELOG.md:29]: I already mentioned the duplicate requests.

**Comment by guidomodarelli** [comment]
TEST :green_circle: 

<details>
  <summary>DISCOVER</summary>

  https://github.com/user-attachments/assets/4bcf5ce1-f531-4908-9cc3-376db0dbb2a7

</details>

https://github.com/user-attachments/assets/94e36cb5-e6d4-41c7-b0b4-901f147f3639


https://github.com/user-attachments/assets/1ae5ee73-8950-4c23-b5d3-88249cdaa8a3

File: plugins/main/public/components/common/wazuh-discover/wz-flyout-discover.tsx
Comment (Desvelao) [inline plugins/main/public/components/common/wazuh-discover/wz-flyout-discover.tsx:1]: I think the usage of "fingerprint" is missing.

File: plugins/main/public/components/common/wazuh-discover/wz-flyout-discover.tsx [hunk 3]
```diff
@@ -144,7 +142,7 @@ const WazuhFlyoutDiscoverComponent = (props: WazuhDiscoverProps) => {
     setIndexPattern(dataSource?.indexPattern);
     fetchData({
       query,
-      dateRange: absoluteDateRange,
+      dateRange: { from: dateRange.from || '', to: dateRange.to || '' },
```
Comment (Desvelao) [inline plugins/main/public/components/common/wazuh-discover/wz-flyout-discover.tsx:145]: **question**: could you elaborate the reason of the default value (empty string)?

File: plugins/main/public/components/common/wazuh-discover/wz-flyout-discover.tsx [hunk 4]
```diff
@@ -307,21 +306,20 @@ const WazuhFlyoutDiscoverComponent = (props: WazuhDiscoverProps) => {
...
+
+                  <EuiFlexGroup
+                    gutterSize='s'
+                    responsive={false}
+                    justifyContent='center'
+                    alignItems='center'
+                  >
+                    <EuiFlexItem grow={false}>
+                      <EuiText size='s'>
+                        {formatUIDate(dateRange?.from)} -{' '}
```
Comment (Desvelao) [inline plugins/main/public/components/common/wazuh-discover/wz-flyout-discover.tsx:318]: **nitpick**: using a relative date such as `Last 24 hours`, this display `Invalid date - Invalid date`:
image

File: plugins/main/public/components/management/cluster/dashboard/dashboard.tsx [hunk 5]
```diff
@@ -95,7 +98,8 @@ const DashboardCT: React.FC<DashboardCTProps> = ({ statusRunning }) => {
   }, [
     JSON.stringify(fetchFilters),
     JSON.stringify(query),
-    JSON.stringify(absoluteDateRange),
+    dateRangeFrom,
+    dateRangeTo,
```
Comment (Desvelao) [inline plugins/main/public/components/management/cluster/dashboard/dashboard.tsx:102]: **nitpick**: I think the usage of "fingerprint" is missing.

File: plugins/main/public/components/overview/amazon-web-services/dashboards/dashboard.tsx [hunk 6]
```diff
@@ -84,7 +84,8 @@ const DashboardAWSComponents: React.FC = ({}) => {
   }, [
     JSON.stringify(fetchFilters),
     JSON.stringify(query),
-    JSON.stringify(absoluteDateRange),
+    dateRangeFrom,
+    dateRangeTo,
```
Comment (Desvelao) [inline plugins/main/public/components/overview/amazon-web-services/dashboards/dashboard.tsx:88]: **nitpick**: I think the usage of "fingerprint" is missing.

File: plugins/main/public/components/overview/compliance-table/compliance-table.tsx [hunk 7]
```diff
@@ -274,14 +274,15 @@ export const ComplianceTable = withAgentSupportModule(props => {
...
+        dateRange: { from: dateRangeFrom, to: dateRangeTo },
       });
     }
   }, [
     dataSource,
     JSON.stringify(searchBarProps.query),
     JSON.stringify(fetchFilters),
-    JSON.stringify(absoluteDateRange),
+    dateRangeFrom,
+    dateRangeTo,
```
Comment (Desvelao) [inline plugins/main/public/components/overview/compliance-table/compliance-table.tsx:285]: **nitpick**: I think the usage of "fingerprint" is missing.

File: plugins/main/public/components/overview/docker/dashboards/dashboard.tsx [hunk 8]
```diff
@@ -84,7 +84,8 @@ const DashboardDockerComponent: React.FC = ({}) => {
   }, [
     JSON.stringify(fetchFilters),
     JSON.stringify(query),
-    JSON.stringify(absoluteDateRange),
+    dateRangeFrom,
+    dateRangeTo,
```
Comment (Desvelao) [inline plugins/main/public/components/overview/docker/dashboards/dashboard.tsx:88]: **nitpick**: I think the usage of "fingerprint" is missing.

File: plugins/main/public/components/overview/fim/dashboard/dashboard.tsx [hunk 9]
```diff
@@ -84,7 +84,8 @@ const DashboardFIMComponent: React.FC = ({}) => {
   }, [
     JSON.stringify(fetchFilters),
     JSON.stringify(query),
-    JSON.stringify(absoluteDateRange),
+    dateRangeFrom,
+    dateRangeTo,
```
Comment (Desvelao) [inline plugins/main/public/components/overview/fim/dashboard/dashboard.tsx:88]: **nitpick**: I think the usage of "fingerprint" is missing.

File: plugins/main/public/components/overview/gdpr/dashboards/dashboard.tsx [hunk 10]
```diff
@@ -85,7 +85,8 @@ const DashboardGDPRComponent: React.FC = () => {
   }, [
     JSON.stringify(fetchFilters),
     JSON.stringify(query),
-    JSON.stringify(absoluteDateRange),
+    dateRangeFrom,
+    dateRangeTo,
```
Comment (Desvelao) [inline plugins/main/public/components/overview/gdpr/dashboards/dashboard.tsx:89]: **nitpick**: I think the usage of "fingerprint" is missing.

File: plugins/main/public/components/overview/github/dashboards/dashboard.tsx [hunk 11]
```diff
@@ -85,7 +85,8 @@ const DashboardGitHubComponent: React.FC = () => {
   }, [
     JSON.stringify(fetchFilters),
     JSON.stringify(query),
-    JSON.stringify(absoluteDateRange),
+    dateRangeFrom,
+    dateRangeTo,
```
Comment (Desvelao) [inline plugins/main/public/components/overview/github/dashboards/dashboard.tsx:89]: **nitpick**: I think the usage of "fingerprint" is missing.

File: plugins/main/public/components/overview/github/panel/config/drilldown-action.tsx
Comment (Desvelao) [inline plugins/main/public/components/overview/github/panel/config/drilldown-action.tsx:1]: **nitpick**: I think the usage of "fingerprint" is missing.

File: plugins/main/public/components/overview/github/panel/config/drilldown-actor.tsx
Comment (Desvelao) [inline plugins/main/public/components/overview/github/panel/config/drilldown-actor.tsx:1]: **nitpick**: I think the usage of "fingerprint" is missing.

File: plugins/main/public/components/overview/github/panel/config/drilldown-organization.tsx
Comment (Desvelao) [inline plugins/main/public/components/overview/github/panel/config/drilldown-organization.tsx:1]: **nitpick**: I think the usage of "fingerprint" is missing.

File: plugins/main/public/components/overview/github/panel/config/drilldown-repository.tsx
Comment (Desvelao) [inline plugins/main/public/components/overview/github/panel/config/drilldown-repository.tsx:1]: **nitpick**: I think the usage of "fingerprint" is missing.

File: plugins/main/public/components/overview/google-cloud/dashboards/dashboard.tsx [hunk 12]
```diff
@@ -83,7 +83,8 @@ const DashboardGoogleCloudComponent: React.FC = () => {
   }, [
     JSON.stringify(fetchFilters),
     JSON.stringify(query),
-    JSON.stringify(absoluteDateRange),
+    dateRangeFrom,
+    dateRangeTo,
```
Comment (Desvelao) [inline plugins/main/public/components/overview/google-cloud/dashboards/dashboard.tsx:87]: **nitpick**: I think the usage of "fingerprint" is missing.

File: plugins/main/public/components/overview/hipaa/dashboards/dashboard.tsx [hunk 13]
```diff
@@ -85,7 +85,8 @@ const DashboardHIPAAComponent: React.FC = () => {
   }, [
     JSON.stringify(fetchFilters),
     JSON.stringify(query),
-    JSON.stringify(absoluteDateRange),
+    dateRangeFrom,
+    dateRangeTo,
```
Comment (Desvelao) [inline plugins/main/public/components/overview/hipaa/dashboards/dashboard.tsx:89]: **nitpick**: I think the usage of "fingerprint" is missing.

File: plugins/main/public/components/overview/malware-detection/dashboard/dashboard.tsx [hunk 14]
```diff
@@ -84,7 +84,8 @@ const DashboardMalwareDetectionComponent: React.FC = ({}) => {
   }, [
     JSON.stringify(fetchFilters),
     JSON.stringify(query),
-    JSON.stringify(absoluteDateRange),
+    dateRangeFrom,
+    dateRangeTo,
```
Comment (Desvelao) [inline plugins/main/public/components/overview/malware-detection/dashboard/dashboard.tsx:88]: **nitpick**: I think the usage of "fingerprint" is missing.

File: plugins/main/public/components/overview/mitre/dashboard/dashboard.tsx [hunk 15]
```diff
@@ -82,7 +82,8 @@ export const DashboardMITRE: React.FC = () => {
   }, [
     JSON.stringify(fetchFilters),
     JSON.stringify(query),
-    JSON.stringify(absoluteDateRange),
+    dateRangeFrom,
+    dateRangeTo,
```
Comment (Desvelao) [inline plugins/main/public/components/overview/mitre/dashboard/dashboard.tsx:86]: **nitpick**: I think the usage of "fingerprint" is missing.

File: plugins/main/public/components/overview/nist/dashboards/dashboard.tsx [hunk 16]
```diff
@@ -85,7 +85,8 @@ const DashboardNIST80053Component: React.FC = () => {
   }, [
     JSON.stringify(fetchFilters),
     JSON.stringify(query),
-    JSON.stringify(absoluteDateRange),
+    dateRangeFrom,
+    dateRangeTo,
```
Comment (Desvelao) [inline plugins/main/public/components/overview/nist/dashboards/dashboard.tsx:89]: **nitpick**: I think the usage of "fingerprint" is missing.

File: plugins/main/public/components/overview/office/dashboard/dashboard.tsx [hunk 17]
```diff
@@ -84,7 +84,8 @@ const DashboardOffice365Component: React.FC = () => {
   }, [
     JSON.stringify(fetchFilters),
     JSON.stringify(query),
-    JSON.stringify(absoluteDateRange),
+    dateRangeFrom,
+    dateRangeTo,
```
Comment (Desvelao) [inline plugins/main/public/components/overview/office/dashboard/dashboard.tsx:88]: **nitpick**: I think the usage of "fingerprint" is missing.

File: plugins/main/public/components/overview/pci/dashboards/dashboard.tsx [hunk 18]
```diff
@@ -86,7 +86,8 @@ const DashboardPCIDSSComponent: React.FC = () => {
   }, [
     JSON.stringify(fetchFilters),
     JSON.stringify(query),
-    JSON.stringify(absoluteDateRange),
+    dateRangeFrom,
+    dateRangeTo,
```
Comment (Desvelao) [inline plugins/main/public/components/overview/pci/dashboards/dashboard.tsx:90]: **nitpick**: I think the usage of "fingerprint" is missing.

File: plugins/main/public/components/overview/server-management-statistics/dashboards/dashboardTabsPanels.tsx [hunk 19]
```diff
@@ -101,7 +104,8 @@ export const DashboardTabsPanels = ({
   }, [
     JSON.stringify(fetchFilters),
     JSON.stringify(query),
-    JSON.stringify(absoluteDateRange),
+    dateRangeFrom,
+    dateRangeTo,
```
Comment (Desvelao) [inline plugins/main/public/components/overview/server-management-statistics/dashboards/dashboardTabsPanels.tsx:108]: **nitpick**: I think the usage of "fingerprint" is missing.

File: plugins/main/public/components/overview/threat-hunting/dashboard/dashboard.tsx [hunk 20]
```diff
@@ -176,7 +176,8 @@ const DashboardTH: React.FC = () => {
     JSON.stringify(query),
     JSON.stringify(pagination),
     JSON.stringify(sorting),
-    JSON.stringify(absoluteDateRange),
+    dateRangeFrom,
+    dateRangeTo,
```
Comment (Desvelao) [inline plugins/main/public/components/overview/threat-hunting/dashboard/dashboard.tsx:180]: **nitpick**: I think the usage of "fingerprint" is missing.

File: plugins/main/public/components/overview/tsc/dashboards/dashboard.tsx [hunk 21]
```diff
@@ -85,7 +85,8 @@ const DashboardTSCComponent: React.FC = () => {
   }, [
     JSON.stringify(fetchFilters),
     JSON.stringify(query),
-    JSON.stringify(absoluteDateRange),
+    dateRangeFrom,
+    dateRangeTo,
```
Comment (Desvelao) [inline plugins/main/public/components/overview/tsc/dashboards/dashboard.tsx:89]: **nitpick**: I think the usage of "fingerprint" is missing.

File: plugins/main/public/components/overview/virustotal/dashboard/dashboard.tsx [hunk 22]
```diff
@@ -86,7 +86,8 @@ const DashboardVT: React.FC = () => {
     isDataSourceLoading,
     JSON.stringify(fetchFilters),
     JSON.stringify(query),
-    JSON.stringify(absoluteDateRange),
+    dateRangeFrom,
+    dateRangeTo,
```
Comment (Desvelao) [inline plugins/main/public/components/overview/virustotal/dashboard/dashboard.tsx:90]: **nitpick**: I think the usage of "fingerprint" is missing.

File: plugins/main/public/components/overview/vulnerabilities/dashboards/overview/dashboard.tsx
Comment (Desvelao) [inline plugins/main/public/components/overview/vulnerabilities/dashboards/overview/dashboard.tsx:1]: **nitpick**: I think the usage of "fingerprint" is missing.

**Comment by Machi3mfl** [comment]
# Tests

## Discover and all sections with a search bar

- Save a query applied in the search bar
- Apply the saved query
- Clear the saved query, the filters, and time filter must be cleared
- Change the filters and check doesn't exists duplicated requests
- Apply filter with absolute time filter, the time filter only must change when the refresh button is clicked
- Use data grid pagination, check the results don't change when the page changes and the filters are the same

**Comment by guidomodarelli** [comment]
# Tests

## Discover and all sections with a search bar

- Render the tabs and check if the requests are duplicated (Exists 2 requests per visualization, approx)

  image

**Comment by Desvelao** [comment]
# wdp-pr-7090

Legend:
:white_circle: no apply
:black_circle:: none
:green_circle:: pass
:yellow_circle:: warning
:red_circle:: fail
:white_circle:: not applicable

## UI

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Render the tabs and verify whether the requests are duplicated (approximately two per visualization). | :green_circle: | :green_circle: | :white_circle: |
| Click the refresh button to update the data grid with the results for the new time range. | :green_circle: | :green_circle: | :white_circle: |
| Utilize the saved query UI handler to save, update, remove, or clear entries | :green_circle: | :black_circle: | :white_circle: |
| Adjust the filters and ensure there are no duplicate requests. | :black_circle: | :black_circle: | :white_circle: |
| Implement a filter using an absolute time range, which should only update upon clicking the refresh button. | :black_circle: | :black_circle: | :white_circle: |
| Implement data grid pagination and verify that the results remain consistent when navigating through pages while the filters stay unchanged. | :black_circle: | :black_circle: | :white_circle: |
| Utilize the auto-refresh feature for time filters and verify the requests. | :black_circle: | :black_circle: | :white_circle: |
| Apply filters using visualizations | :black_circle: | :black_circle: | :white_circle: |
| Create a report and verify the results displayed in the PDF. | :black_circle: | :black_circle: | :white_circle: |
| Check cluster, endpoints summary, statistics | :black_circle: | :black_circle: | :white_circle: |
| For vulnerability detection, examine filters and review saved queries | :black_circle: | :black_circle: | :white_circle: |
| Pin agent and unpin agent in the dashboard, check the requests | :black_circle: | :black_circle: | :white_circle: |

**Details**
<details>
<summary>:green_circle: Render the tabs and verify whether the requests are duplicated (approximately two per visualization).</summary>

Chrome - :green_circle:

image
Firefox - :green_circle:

image

Safari - :black_circle:

</details>

<details>
<summary>:green_circle: Click the refresh button to update the data grid with the results for the new time range.</summary>

Chrome - :green_circle:
image
image
image

Firefox - :green_circle:
image
image
image

Safari - :black_circle:

</details>

<details>
<summary>:green_circle: Utilize the saved query UI handler to save, update, remove, or clear entries</summary>

Chrome - :green_circle:
image

Firefox - :green_circle:
image

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Adjust the filters and ensure there are no duplicate requests.</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:green_circle: Implement a filter using an absolute time range, which should only update upon clicking the refresh button.</summary>

Chrome - :green_circle:
image

Firefox - :green_circle:
image

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Implement data grid pagination and verify that the results remain consistent when navigating through pages while the filters stay unchanged.</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Utilize the auto-refresh feature for time filters and verify the requests.</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Apply filters using visualizations</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Create a report and verify the results displayed in the PDF.</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Check cluster, endpoints summary, statistics</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: For vulnerability detection, examine filters and review saved queries</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Pin agent and unpin agent in the dashboard, check the requests</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

**Review by Machi3mfl — APPROVED** [review]

**Review by Desvelao — APPROVED** [review]
LGTM!

**Pull request merged**